### PR TITLE
Fix #1905: Disallow @js.native on Scala classes

### LINKED ISSUE #1905: Using @js.native on a non-js.Any should be an error
It seems I forgot to report an error in this case:
```scala
@js.native
class Foo
```
which is very disturbing.

### PR DESCRIPTION
**Pull request by gzm0**

### CODE REVIEW

**Comment by gzm0** [comment]
Review by @sjrd

File: compiler/src/test/scala/org/scalajs/core/compiler/test/JSInteropTest.scala [hunk 1]
```diff
@@ -198,25 +233,22 @@ class JSInteropTest extends DirectTest with TestHelpers {
   def onlyJSRawTraits: Unit = {
 
     """
-    @js.native
     trait A
```
Comment (sjrd) [inline compiler/src/test/scala/org/scalajs/core/compiler/test/JSInteropTest.scala]: lol ^^

**Comment by sjrd** [comment]
LGTM

File: compiler/src/test/scala/org/scalajs/core/compiler/test/JSInteropTest.scala (hunk 1 above)
Comment (gzm0) [inline compiler/src/test/scala/org/scalajs/core/compiler/test/JSInteropTest.scala]: I was wondering for quite a while whether this was intended and what it should mean...

**Pull request merged**